snapcraft/internal/errors.py

```diff
@@ -166,6 +166,14 @@ class MissingGadgetError(SnapcraftError):
         'https://github.com/snapcore/snapd/wiki/Gadget-snap')
 
 
+class PluginOutdatedError(SnapcraftError):
+
+    fmt = 'This plugin is outdated: {message}'
+
+    def __init__(self, message):
```

Comment:
you only really need to define `fmt` in these cases, `__init__` would do the right thing given `kwargs` in the super class.